### PR TITLE
#6648: Update iframes from page editor without full page reload

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Closes https://github.com/pixiebrix/pixiebrix-extension/issues/6648
- Includes  https://github.com/pixiebrix/webext-messenger/pull/209

Pretty straightforward change both in the Messenger and here. It comes "free" from the browser because "message all frames" is actually the default behavior of `tabs.sendMessage`

## Demo

Test URL: https://ephiframe.vercel.app/Outer?iframe=https://extra-ephiframe.vercel.app/Cross-domain-iframe%3Fiframe%3D./nested

Supports:

- Added after permission change
- Removed after permission change
- Updated after mod change
- Removed after mod removal

https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/510317ff-b4bc-4cac-8151-3f8a40268965

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/contentScript/messenger/api.ts [hunk 1]
```diff
@@ -37,8 +37,8 @@ export const getAttributeExamples = getMethod("GET_ATTRIBUTE_EXAMPLES");
 export const runBlock = getMethod("RUN_SINGLE_BLOCK");
 export const runRendererBlock = getMethod("RUN_RENDERER_BLOCK");
 
-export const clearDynamicElements = getMethod("CLEAR_DYNAMIC_ELEMENTS");
-export const updateDynamicElement = getMethod("UPDATE_DYNAMIC_ELEMENT");
+export const clearDynamicElements = getNotifier("CLEAR_DYNAMIC_ELEMENTS");
+export const updateDynamicElement = getNotifier("UPDATE_DYNAMIC_ELEMENT");
```
Comment (fregante) [inline src/contentScript/messenger/api.ts:41]: Not necessary but it looks like most places don't await these calls or are unlikely to throw.

File: src/pageEditor/PanelContent.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ const cleanUpStarterBrickForElement = (
     return;
   }
 
-  removeInstalledExtension(thisTab, element.uuid);
+  removeInstalledExtension(allFramesInThisTab, element.uuid);
```
Comment (fregante) [inline src/pageEditor/PanelContent.tsx:63]: I don't recognize this file. If this extension isn't expected to ever be in a frame, I can undo this change.

File: src/pageEditor/protocol.ts [hunk 3]
```diff
@@ -38,7 +38,7 @@ async function onNavigation(target: Target): Promise<void> {
 }
 
 function onEditorClose(): void {
-  resetTab(thisTab);
+  resetTab(allFramesInThisTab);
```
Comment (fregante) [inline src/pageEditor/protocol.ts:41]: Also verified

File: src/pageEditor/PanelContent.tsx (hunk 2 above)
Comment (grahamlangford) [reply src/pageEditor/PanelContent.tsx:63]: I don't think we use iframes anywhere in the pageEditor. Nothing comes up from a simple search either. @BLoe, can you confirm?

**Review by grahamlangford — APPROVED** [review]
Conditionally approved. Let's get Ben's feedback on the `PanelContent` change.

File: src/pageEditor/PanelContent.tsx (hunk 2 above)
Comment (fregante) [reply src/pageEditor/PanelContent.tsx:63]: `thisTab` refers the document the page editor is connected to. `allFramesInThisTab` includes all of its iframes. They don't live in the page editor itself

**Pull request merged**